packages/origin-ui-irec-core/src/components/table/DeviceTable.tsx

```diff
@@ -0,0 +1,317 @@
+    const environment = useSelector(getEnvironment);
+    const exchangeDepositAddress = useSelector(getExchangeDepositAddress);
+    const [showRequestCertModal, setShowRequestCertModal] = useState(false);
+    const [producingDeviceForModal, setProducingDeviceForModal] = useState(null);
+    const dispatch = useDispatch();
+
+    // Check if this needed ?
+    function enrichProducingDeviceData(): IEnrichedDeviceData[] {
+        const enriched: IEnrichedDeviceData[] = [];
+        for (const device of devices) {
+            enriched.push({
+                device,
+                // check if it is true
+                organizationName: device.registrantOrganization,
+                locationText: getDeviceLocationText(device)
+            });
+        }
+
+        return enriched;
+    }
+
+    async function getPaginatedData({
+        requestedPageSize,
+        offset,
+        requestedFilters
+    }: IPaginatedLoaderHooksFetchDataParameters): Promise<IPaginatedLoaderFetchDataReturnValues> {
+        const enrichedDeviceData = await enrichProducingDeviceData();
+
+        const includedStatuses = props.includedStatuses || [];
+
+        const filteredEnrichedDeviceData = enrichedDeviceData.filter(
+            (record) =>
+                checkRecordPassesFilters(
+                    record,
+                    requestedFilters,
+                    configuration.deviceTypeService
+                ) &&
+                (!props.owner ||
+                    // check if its true later
+                    record?.device.registrantOrganization === user?.organization?.id.toString()) &&
+                (includedStatuses.length === 0 || includedStatuses.includes(record.device.status))
+        );
+
+        const total = filteredEnrichedDeviceData.length;
+
+        const paginatedData = filteredEnrichedDeviceData.slice(offset, offset + requestedPageSize);
+
+        return {
+            paginatedData,
+            total
+        };
+    }
+
+    const { paginatedData, loadPage, total, pageSize } = usePaginatedLoaderFiltered<
+        IEnrichedDeviceData
+    >({
+        getPaginatedData
+    });
+
+    const { t } = useTranslation();
+
+    useEffect(() => {
+        loadPage(1);
+    }, [user, devices]);
+
+    function viewDevice(rowIndex: number) {
+        const device = paginatedData[rowIndex].device;
+
+        setDetailViewForDeviceId(device.id);
+    }
+
+    async function requestCerts(rowIndex: string) {
+        const producingDevice = paginatedData[rowIndex].device;
+
+        if (producingDevice.status !== DeviceStatus.Active) {
+            return showNotification(
+                `You can only request certificates for devices with status ${DeviceStatus.Active}.`,
+                NotificationType.Error
+            );
+        }
+
+        setProducingDeviceForModal(producingDevice);
+        setShowRequestCertModal(true);
+    }
+
+    async function approve(rowIndex: string) {
```

Comment:
In general, the final I-REC integration will not have approval functionality steered by the platform admin. This action will be taken by the I-REC organization.

For now though, since the I-REC integration is not ready, we need a way to approve devices hence the local-irec pacakge has that functionality. Please use it https://github.com/energywebfoundation/origin/blob/ec5aa08791af0abfe3b1ee059657518dff68c05e/packages/origin-device-registry-irec-local-api/src/device/device.controller.ts#L106